### PR TITLE
MM-31064: Simplify indentation

### PR DESCRIPTION
**Pull request by agnivade**
Reduce indentation where possible.

```release-note
NONE
```

Command ran to verify:
golangci-lint run --disable-all --enable golint --max-issues-per-linter=10000 --max-same-issues=100000 ./... | grep "block ends with a return state"

https://mattermost.atlassian.net/browse/MM-31064

### CODE REVIEW

**Review by jespino — APPROVED** [review]
I love this thing! :heart:

**Comment by jespino** [comment]
Do we need to add the check to our golang-ci configuration to avoid reintroducing errors here?

**Comment by agnivade** [comment]
I will check if it's possible to configure a single golint rule. The idea for this mainly came out of a style guide that the server team has created. But we need to first make the entire codebase consistent.

File: app/webhook.go [hunk 1]
```diff
@@ -738,17 +737,17 @@ func (a *App) HandleIncomingWebhook(hookId string, req *model.IncomingWebhookReq
...
+			case errors.As(result2.NErr, &nfErr):
 				return model.NewAppError("HandleIncomingWebhook", "web.incoming_webhook.channel.app_error", nil, nfErr.Error(), http.StatusNotFound)
 			default:
-				return model.NewAppError("HandleIncomingWebhook", "web.incoming_webhook.channel.app_error", nil, result.NErr.Error(), http.StatusInternalServerError)
+				return model.NewAppError("HandleIncomingWebhook", "web.incoming_webhook.channel.app_error", nil, result2.NErr.Error(), http.StatusInternalServerError)
 			}
 		} else {
-			channel = result.Data.(*model.Channel)
+			channel = result2.Data.(*model.Channel)
 		}
```
Comment (isacikgoz) [inline app/webhook.go:751]: Can we also remove this `else`?

**Review by isacikgoz — COMMENTED** [review]
Awesome improvement for our necks 😛 Just spotted a thing that might escaped.

File: app/webhook.go (hunk 1 above)
Comment (agnivade) [reply app/webhook.go:751]: Yep, nice one.

**Comment by agnivade** [comment]
@isacikgoz - Need your help a bit with this one. :sweat_smile:  There are some pending issues in the EE repo to be also fixed before we can merge this. Would you be open to give a hand ?

**Review by isacikgoz — APPROVED** [review]
LGTM 💯

**Pull request merged**